projects/storefrontlib/src/cms-components/misc/qualtrics/qualtrics-loader.service.ts

```diff
@@ -1,78 +1,131 @@
-import { Injectable } from '@angular/core';
+import {
+  Injectable,
+  isDevMode,
+  Renderer2,
+  RendererFactory2,
+} from '@angular/core';
 import { WindowRef } from '@spartacus/core';
-import { BehaviorSubject, fromEvent, Observable, of } from 'rxjs';
-import { distinctUntilChanged, filter, switchMap, tap } from 'rxjs/operators';
-import { QualtricsConfig } from './config/qualtrics-config';
+import { fromEvent, Observable, of } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
 
+/**
+ * Service to integration Qualtrics.
+ *
+ * The integration observes the Qualtrics API, and when available, it runs the QSI API
+ * to let Qualtrics evaluate the application.
+ *
+ * The service supports an additional _hook_ (`isDataLoaded()`) that can be used to load application
+ * data before pulling the QSI API. This is benefitial in a single page application when additional
+ * data is required befor the Qualtrics _creatives_ run.
+ *
+ * This service also supports the creation of the Qualtrics deployment script. This is optional, as
+ * the script can be added in various ways.
```

Comment:
Would it be beneficial to list here what are all the options to add the script?
Or is this something that should be in the docs repo?

Reply:
I considered it, but thought it's better to do it in the docs. It requires quite a bit of explanation.